### PR TITLE
feature: Add voltage divider device

### PR DESCRIPTION
**Pull request by giulioungaretti**

### CODE REVIEW

File: qcodes/instrument_drivers/devices.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        Returns:
+            number: value at which was set at the sample
+        """
+        return self.v1.get() / self.division_value
+
+    def get_instrument_value(self) -> Union[int, float]:
+        """
+        Returns:
+            number: value at which the attached paraemter is (i.e. does not
+                    account for the scaling)
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/devices.py]: This should un-indent to the level of number to render correctly

File: qcodes/instrument_drivers/devices.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+from typing import Union
+
+from qcodes import Parameter, StandardParameter
+
+
+class VoltageDivider(Parameter):
+    """
+    Resitive voltage divider
+
+    To be used when you use a voltage divider to measure a parameter.
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/devices.py]: Are you measuring the parameter. The example sounds more like you are setting an output?

Output of dac is X -> Voltage divider -> x/10 to actual Experiment

Comment (giulioungaretti) [inline qcodes/instrument_drivers/devices.py]: I think it's fine like this, you can also get values.

**Comment by giulioungaretti** [comment]
@QCoDeS/core  moar opinions, else I will just merge and let people try ?

**Comment by jenshnielsen** [comment]
:+1: examples of using it both in input and output would probably make it more clear how it works but that can wait

**Comment by MerlinSmiles** [comment]
Just want to throw two things in here:
https://github.com/QCoDeS/Qcodes/blob/master/qcodes/instrument_drivers/stanford_research/SR560.py
and #57 

Maybe its more general then adding devices for all kinds of conversions...?

**Comment by jenshnielsen** [comment]
I think it makes sense to move the VoltageParameter and also the CurrentParameter from https://github.com/QCoDeS/Qcodes/blob/master/qcodes/instrument_drivers/ithaco/Ithaco_1211.py#L7 here to this location they dont really seem to belong in a specific instrument driver

**Comment by giulioungaretti** [comment]
@jenshnielsen doc wise, I am not sure I understand. there are both sets and gets in the examples :D 
@MerlinSmiles @jenshnielsen  yeah it may make sense, but let's postpone the dicussion untill we have a working dataset. It's a lot better to repeat some code than to make the wrong abstractions IMHO :unicorn:

File: qcodes/instrument_drivers/devices.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            label=self.label,
+            unit=self.v1.unit,
+            metadata=self.v1.metadata)
+
+        # extend metadata
+        self._meta_attrs.extend(['v1', 'division_value'])
+
+    def set(self, value: Union[int, float]) -> None:
+        instrument_value = value * self.division_value
+        self._save_val(value)
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/devices.py]: what happens if instrument does not have a set function

Comment (giulioungaretti) [inline qcodes/instrument_drivers/devices.py]: well it's going to raise an exception :D as expected :D

**Comment by giulioungaretti** [comment]
@jenshnielsen  more feedback, I would merge other wise :D

**Comment by jenshnielsen** [comment]
IMHO the current documentation is still not very clear in the distinction between using it as an input or output device. I had to read it several times to understand how it was ment to be used in the two modes and what refers to what but perhaps it's more clear to others. Otherwise 👍

File: qcodes/instrument_drivers/devices.py (hunk 3 above)
Comment (nulinspiratie) [inline qcodes/instrument_drivers/devices.py]: How about setting the parameter's has_set to equal that of the original parameter?

Comment (giulioungaretti) [inline qcodes/instrument_drivers/devices.py]: @nulinspiratie, right now this is the error you 'd get.
```python
/src/Qcodes/qcodes/instrument/parameter.py in no_setter(*args, **kwargs)
    676
    677 def no_setter(*args, **kwargs):
--> 678     raise NotImplementedError('This Parameter has no setter defined.')


NotImplementedError: ('This Parameter has no setter defined.', 'setting yolo_parabola to 0')

```
The difference why using  has_set is that one gets the real error i.e. that the underlying parameter has no set :D 

Does that make sense ?

Comment (nulinspiratie) [inline qcodes/instrument_drivers/devices.py]: Ah yeah that would give a clearer error message

**Pull request merged**